cgi-bin/LJ/Talk.pm

```diff
@@ -1515,40 +1536,7 @@ sub talkform {
             }
         : 0,
 
-        remote => $remote
-        ? {
-            icons_url => $remote ? $remote->allpics_base : '',
-            icons     => \@icons,
-
-            user            => $remote->user,
-            display_name    => LJ::ehtml( $remote->display_name ),
-            ljuser          => $remote->ljuser_display,
-            openid_identity => $remote->openid_identity,
-
-            allowed  => !$journalu->does_not_allow_comments_from($remote),
-            banned   => $journalu->has_banned($remote),
-            screened => (
-                       $journalu->has_autoscreen($remote)
-                    || $screening eq 'A'
-                    || ( $screening eq 'R' && !$remote->is_validated )
-                    || ( $screening eq 'F' && !$journalu->trusts($remote) )
-            ),
-
-            can_unscreen_parent => (
-                       $parpost->{state}
-                    && $parpost->{state} eq "S"
-                    && LJ::Talk::can_unscreen( $remote, $journalu, $entry->poster )
-            ),
-            can_manage_community => $journalu->is_community
-                && $remote
-                && $remote->can_manage($journalu),
-
-            can_use_userpic_select   => $remote->can_use_userpic_select && ( scalar(@icons) > 0 ),
-            iconbrowser_keywordorder => $remote->iconbrowser_keywordorder ? "true" : "false",
-            iconbrowser_metatext     => $remote->iconbrowser_metatext ? "true" : "false",
-            iconbrowser_smallicons   => $remote->iconbrowser_smallicons ? "true" : "false",
-            }
-        : 0,
+        remote  => $remote ? $remote : 0,
```

Comment:
Why 0? Shouldn't this be undef, if it's "there is no remote"? Or does TT want a 0 for some reason... and if it's just undef, we should just say `remote => $remote`.

Reply:
This one is a question for @nfagerlund as that bit of cose is his.